java/arcs/core/analysis/RecipeGraph.kt

```diff
@@ -130,13 +138,29 @@ class RecipeGraph(recipe: Recipe) {
         }
 
         /** A node representing a particle. */
-        data class Particle(val particle: Recipe.Particle) : Node() {
-            override fun toString() = "[p:${particle.spec.name}]"
+        data class Particle(
+            val particle: Recipe.Particle,
```

Comment:
To avoid: `chicken.chicken` one could have:
```
data class ParticleNode(
    val recipeParticle: Recipe.Particle,
    (...)
```

Up to your consideration, use your best judgement.

Reply:
This tends to be used with a qualified type name (i.e. `Node.Particle`, not just `Particle`), and named like so: `particleNode: Node.Particle`. `particleNode.particle` isn't so bad imo.